agent/php_txn.c

```diff
@@ -914,17 +916,17 @@ nr_status_t nr_php_txn_begin(const char* appnames,
     nr_php_txn_log_error_dt_on_tt_off();
   }
 
-#if ZEND_MODULE_API_NO >= ZEND_8_1_X_API_NO
-  if (nr_php_ini_setting_is_set_by_user("opcache.enable")
-      && NR_PHP_PROCESS_GLOBALS(preload_framework_library_detection)) {
-    nr_php_user_instrumentation_from_opcache(TSRMLS_C);
-  }
-#else
-  if (nr_php_ini_setting_is_set_by_user("opcache.preload")
-      && NR_PHP_PROCESS_GLOBALS(preload_framework_library_detection)) {
-    nr_php_user_instrumentation_from_opcache(TSRMLS_C);
+  /*
+   * Only try to instrument preloaded opcache scripts when opcache enabled and
+   * preload is nt null.  If an INI value does not exist, INI_INT returns 0 and
+   * INI_STR returns NULL.
+   */
+  if (NR_PHP_PROCESS_GLOBALS(preload_framework_library_detection)) {
+    if ((1 == INI_INT("opcache.enable"))
+        && (NULL != INI_STR("opcache.preload"))) {
```

Comment:
Regarding INI_STR

See https://github.com/php/php-src/blob/master/Zend/zend_ini.c and https://github.com/php/php-src/blob/master/Zend/zend_ini.h for details about the INI_STR function.  

If a value is not set ([which is the default](https://github.com/php/php-src/blob/master/php.ini-production#L1927)) INI_STR will return NULL so it is equivalent to the functionality of the `nr_php_ini_setting_is_set_by_user` function here: https://github.com/newrelic/newrelic-php-agent/blob/main/agent/php_nrini.c#L3285 which uses zend_get_configuration_directive  (also in zend.c) to just checks if the value is set as well.  Neither method checks for empty string as user could also set to empty string. INI_STR has the advantage of being native PHP code, so we don't need to toggle between PHP versions like `nr_php_ini_setting_is_set_by_user` does and it also had fewer variable sets and function calls so would be more performant.  However since they are both equivalent and it does affect PHP 7.4, 8.0 and 8.1, agree that we can just continue to use `nr_php_ini_setting_is_set_by_user`.

Reply:
> If a value is not set ([which is the default](https://github.com/php/php-src/blob/master/php.ini-production#L1927)) INI_STR will return NULL

This is not what I saw when I tested this with PHP 8.0. I have this in my `php.ini`:
```
ubuntu@php:~/code/php-agent$ grep opcache.preload /opt/nr/etc/php-8.0/php.ini
; http://php.net/opcache.preload
;opcache.preload=
; http://php.net/opcache.preload_user
;opcache.preload_user=
```

This is what is displayed by `php -i`:
```
ubuntu@php:~/code/php-agent$ php -i | grep -E 'opcache.enable|opcache.preload'
opcache.enable => On => On
opcache.enable_cli => On => On
opcache.enable_file_override => Off => Off
opcache.preload => no value => no value
opcache.preload_user => no value => no value
```

So PHP is using default for `opcache.preload`. However `INI_STR("opcache.preload")` returns a value which is not `NULL` which results in a call to `nr_php_user_instrumentation_from_opcache` and that yields the warning:
```
warning: User instrumentation from opcache: missing 'scripts' key in status information
```

Reply:
Moreover `nr_php_ini_setting_is_set_by_user("opcache.preload")` returns false in the above scenario which does not result in a call to `nr_php_user_instrumentation_from_opcache`.

Reply:
Please see: [fe66b6f](https://github.com/newrelic/newrelic-php-agent/pull/595/commits/fe66b6fa6fda4bfc0253eb950ff1cf6be0a63994)

Reply:
Yes, this reads and works better. Thank you.